docs/extend/modular-resources/create.md

```diff
@@ -0,0 +1,511 @@
+---
+title: "Code your own modules to create custom resources"
+linkTitle: "Create"
+weight: 20
+type: "docs"
+tags: ["server", "rdk", "extending viam", "modular resources", "components", "services"]
+description: "Use the Viam module system to implement custom resources that can be included in any Viam-powered robot."
+no_list: true
+---
+
+The Viam module system allows you to integrate custom {{< glossary_tooltip term_id="resource" text="resources" >}}, [components](/components/), and [services](/services/), into any robot running on Viam.
+
+A common use case for modular resources is to create a new model that implements an existing Viam API.
+However, you can also create and expose new API types using modular resources.
+
+{{% alert title="Modules vs. modular resources" color="tip" %}}
+
+A configured *module* can make one or more *modular resources* available for configuration.
+
+{{% /alert %}}
+
+## Create a custom modular resource
+
+To create your own modular resource, code a module in Go or Python using the module support libraries provided by [Viam's SDKs](/program/apis/) that implements at least one new model or type of {{< glossary_tooltip term_id="resource" text="resource" >}}:
+
+{{< tabs >}}
+{{% tab name="Define a New Model of a Built-In Resource Type" %}}
+
```

Comment:
```suggestion
{{% tab name="New Model" %}}
Define a new model of a built-in resource type:

```
[opt] Realized these tab names are so long that they show up stacked oddly on many screens.